EndToEndChannelLeakTest  test failure

**Problem description**
EndToEndChannelLeakTest  fails sporadically.
```
io.pravega.test.integration.endtoendtest.EndToEndChannelLeakTest > testDetectChannelLeakMultiReader FAILED
    java.lang.AssertionError: expected:<5> but was:<4>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at io.pravega.test.integration.endtoendtest.EndToEndChannelLeakTest.testDetectChannelLeakMultiReader(EndToEndChannelLeakTest.java:207)
```
**Problem location**
EndToEndChannelLeakTest.java

**Suggestions for an improvement**
Fix the race condition in the test.

--- a/test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndChannelLeakTest.java
+++ b/test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndChannelLeakTest.java
@@ -177,43 +177,40 @@ public class EndToEndChannelLeakTest {
         ConnectionFactoryImpl connectionFactory = new ConnectionFactoryImpl(ClientConfig.builder().build());
         @Cleanup
         ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE, controller, connectionFactory);
-
-        assertEquals(0, connectionFactory.getActiveChannelCount());
+        int expectedChannelCount = 0; // open socket count.
+        assertEquals(expectedChannelCount, connectionFactory.getActiveChannelCount());
         
-        //Create a writer.
+        //Create a writer and write an event.
         @Cleanup
         EventStreamWriter<String> writer = clientFactory.createEventWriter(STREAM_NAME, new JavaSerializer<>(),
                 EventWriterConfig.builder().build());
-        
-        assertEquals(0, connectionFactory.getActiveChannelCount());
-        //Write an event.
         writer.writeEvent("0", "zero").get();
-        assertEquals(1, connectionFactory.getActiveChannelCount());
+
+        expectedChannelCount += 1; // connection to segment 0.
+        assertEquals(expectedChannelCount, connectionFactory.getActiveChannelCount());
         
         @Cleanup
         ReaderGroupManager groupManager = new ReaderGroupManagerImpl(SCOPE, controller, clientFactory,
                 connectionFactory);
-        assertEquals(1, connectionFactory.getActiveChannelCount());
+        assertEquals(expectedChannelCount, connectionFactory.getActiveChannelCount()); // no changes expected.
       
         groupManager.createReaderGroup(READER_GROUP, ReaderGroupConfig.builder().disableAutomaticCheckpoints()
                 .stream(Stream.of(SCOPE, STREAM_NAME)).build());
-        //create a reader.
+
+        //create a reader and read an event.
         @Cleanup
         EventStreamReader<String> reader1 = clientFactory.createReader("readerId1", READER_GROUP, serializer,
                 ReaderConfig.builder().build());
-
-        //Total 5 sockets are open at this point : Writer has 1 connection to segment 0 of stream +
-        //Reader has 4 connections to the reader group (read, write, metadata, and conditionalUpdates)
-        assertEquals(5, connectionFactory.getActiveChannelCount());
-        int channelCount = 5;
-
-        //Read an event.
+        //Creating a reader spawns a revisioned stream client which opens 4 sockets ( read, write, metadataClient and conditionalUpdates).
+        expectedChannelCount += 4;
         EventRead<String> event = reader1.readNextEvent(10000);
+        //reader creates a new connection to the segment 0;
+        expectedChannelCount += 1;
+
         assertNotNull(event);
         assertEquals("zero", event.getEvent());
         //+1 socket to segment 0 of the stream.
-        assertEquals(channelCount + 1, connectionFactory.getActiveChannelCount());
-        channelCount = channelCount + 1;
+        assertEquals(expectedChannelCount, connectionFactory.getActiveChannelCount());
 
         // scale
         Stream stream = new StreamImpl(SCOPE, STREAM_NAME);
@@ -224,36 +221,38 @@ public class EndToEndChannelLeakTest {
         Boolean result = controller.scaleStream(stream, Collections.singletonList(0), map, executor).getFuture().get();
         assertTrue(result);
         //No changes to the channel count.
-        assertEquals(channelCount, connectionFactory.getActiveChannelCount());
+        assertEquals(expectedChannelCount, connectionFactory.getActiveChannelCount());
 
         //Write more events.
         writer.writeEvent("0", "one").get();
         writer.writeEvent("0", "two").get();
         writer.writeEvent("1", "three").get();
 
-        //2 new connections(+3 connections to the segments 1,2,3 after scale by the writer,
+        //2 new connections are opened.(+3 connections to the segments 1,2,3 after scale by the writer,
         // -1 connection to segment 0 which is sealed.)
-        assertEquals(channelCount + 2, connectionFactory.getActiveChannelCount());
-        channelCount = channelCount + 2;
+        expectedChannelCount += 2;
+        assertEquals(expectedChannelCount, connectionFactory.getActiveChannelCount());
 
         //Add a new reader
         @Cleanup
         EventStreamReader<String> reader2 = clientFactory.createReader("readerId2", READER_GROUP, serializer,
                 ReaderConfig.builder().build());
-        //Creation of a reader will add 4 more connections details similar to the above comment.
-        assertEquals(channelCount + 4, connectionFactory.getActiveChannelCount());
-        channelCount = channelCount + 4;
+        //Creating a reader spawns a revisioned stream client which opens 4 sockets ( read, write, metadataClient and conditionalUpdates).
+        expectedChannelCount += 4;
 
         event = reader1.readNextEvent(10000);
         assertNotNull(event);
+
         //+1 connection (-1 since segment 0 of stream is sealed + 2 connections to two segments of stream (there are
         // 2 readers and 3 segments and the reader1 will be assigned 2 segments))
-        assertEquals(channelCount + 1, connectionFactory.getActiveChannelCount());
-        channelCount = channelCount + 1;
+        expectedChannelCount += 1;
+        assertEquals(expectedChannelCount, connectionFactory.getActiveChannelCount());
 
         event = reader2.readNextEvent(10000);
         assertNotNull(event);
+
         //+1 connection (a new connection to the remaining stream segment)
-        assertEquals(channelCount + 1, connectionFactory.getActiveChannelCount());
+        expectedChannelCount += 1;
+        assertEquals(expectedChannelCount, connectionFactory.getActiveChannelCount());
     }
 }

